### PR TITLE
Updates for Unified IMX6/RPi/Jetson Build System

### LINKED ISSUE #71: do_roofts fails on some component
> ERROR: var-dev-update-full-image-1.0-r0 do_rootfs: Unable to install packages. Command '/ephemeral/yocto-ornl-dunfell/build_ornl/tmp/work/var_som_mx6_ornl-fslc-linux-gnueabi/var-dev-update-full-image/1.0-r0/recipe-sysroot-native/usr/bin/opkg --volatile-cache -f /ephemeral/yocto-ornl-dunfell/build_ornl/tmp/work/var_som_mx6_ornl-fslc-linux-gnueabi/var-dev-update-full-image/1.0-r0/opkg.conf -t /ephemeral/yocto-ornl-dunfell/build_ornl/tmp/work/var_som_mx6_ornl-fslc-linux-gnueabi/var-dev-update-full-image/1.0-r0/temp/ipktemp/ -o /ephemeral/yocto-ornl-dunfell/build_ornl/tmp/work/var_som_mx6_ornl-fslc-linux-gnueabi/var-dev-update-full-image/1.0-r0/rootfs  --force_postinstall --prefer-arch-to-version   install base-passwd bind-utils default-eth0 dotnet-core dtc elp-h264 flex fsl-rc-local gcc git gps-utils gpsd gst-interpipe gstd gstreamer1.0-plugins-ugly htop i2c-tools imx-test iotop iperf3 kernel-devicetree kernel-image ksz-initscripts libgps libsodium libsodium-dev libtool libxml2-dev libxslt-dev m4 make mfgtest minicom nano net-tools networkmanager nload nodejs ntp ntp-bin openssl openssl-bin packagegroup-base-extended packagegroup-cockpit packagegroup-core-boot packagegroup-core-full-cmdline packagegroup-core-sdk packagegroup-core-ssh-openssh packagegroup-core-standalone-sdk-target packagegroup-core-tools-debug packagegroup-fsl-gstreamer1.0 packagegroup-fsl-gstreamer1.0-full packagegroup-fsl-tools-benchmark packagegroup-fsl-tools-gpu packagegroup-fsl-tools-gpu-external packagegroup-fsl-tools-testapps packagegroup-imx-tools-audio packagegroup-tools-bluetooth perf pkgconfig postinstall powertop psplash python3 python3-bottle python3-dev python3-future python3-lxml python3-mavproxy python3-netifaces python3-numpy python3-pexpect python3-pip python3-protobuf python3-pymavlink python3-pynmea2 python3-pyserial python3-pytz python3-pyyaml python3-requests python3-uptime python3-urllib3 run-postinsts screen shadow strace sudo swupdate swupdate-www systemd-analyze tcpdump u-boot-variscite v4l-utils vsdbg webcam-tools x264 x265' returned 255:

This causes a `var-dev-update-full-image` build to fail, although most items are successfully installed.  Current workaround is to ignore the exit code in the Makefile to continue with the build steps, but this is a bad hack.

### PR DESCRIPTION
**Pull request by bvacaliuc**
This has necessary fixes to the build system and documentation.  I believe it incorporates the changes to the unified rpi and nvidia jetson targets as well as functioning for the variscite board.

NOTE: it does *not* produce a working `.swu` file at this time.  However, nothing in the dunfell has produced this; the last working `.swu` files were on thud.

### CODE REVIEW

**Comment by bvacaliuc** [comment]
This PR supercedes #55

**Comment by bvacaliuc** [comment]
There is currently a problem with `.swu` updates on this branch.  Here is the status:

- [x] flash to microSD and boot
- [x] install to emmc (two partitions)
- [ ] use `.swu` file to update rootfs on emmc

File: sources/meta-ornl/conf/machine/include/ornl.inc [hunk 1]
```diff
@@ -24,6 +24,7 @@ UBI_VOLNAME  = "rootfs"
 
 # Variscite BSP default providers
 PREFERRED_PROVIDER_virtual/kernel ?= "linux-variscite"
+# FIXME: this breaks thud when it uses 4.9.88
```
Comment (waddle77) [inline sources/meta-ornl/conf/machine/include/ornl.inc:27]: Does any of our projects still use Thud?  Can we EOL Thud just as the official Yocto Project has?

Comment (waddle77) [inline sources/meta-ornl/conf/machine/include/ornl.inc:27]: Does any of our projects still use Thud? Can we EOL Thud just as the official Yocto Project has?

File: sources/meta-ornl/recipes-core/ksz-initscripts/ksz-initscripts_1.0.bb [hunk 2]
```diff
@@ -36,4 +36,4 @@ do_install() {
     fi
 }
 
-COMPATIBLE_MACHINE = "var-som-mx6-ornl"
\ No newline at end of file
+COMPATIBLE_MACHINE = "var-som-mx6-ornl"
```
Comment (waddle77) [inline sources/meta-ornl/recipes-core/ksz-initscripts/ksz-initscripts_1.0.bb:39]: What changed here?

File: sources/meta-ornl/recipes-variscite/images/var-dev-update-full-image.bb [hunk 3]
```diff
@@ -8,10 +8,25 @@ require recipes-core/images/ornl-dev-image.bb
...
-    kernel-image \
-    kernel-devicetree \
-    ksz-initscripts \
-    u-boot-variscite \
+	imx-test \
+	kernel-devicetree \
+	kernel-image \
+	ksz-initscripts \
+	mfgtest \
+	packagegroup-core-full-cmdline \
```
Comment (waddle77) [inline sources/meta-ornl/recipes-variscite/images/var-dev-update-full-image.bb:16]: Where were these packagegroups located before this?  Were they not being built in?

**Review by waddle77 — APPROVED** [review]
These look good to me, I just have a few questions.

**Comment by bvacaliuc** [comment]
see #69 for current build issue

**Pull request merged**